src/pool/poolClient.ts

```diff
@@ -119,7 +119,9 @@ class PoolState {
       exchangeRateCurrent,
       ...pooledToken,
     }
-
+  }
+  public async exchangeRateAtBlock(l1Token:string, blockTag:number){
+    return this.contract.callStatic.exchangeRateCurrent(l1Token,{ blockTag } as any);
```

Comment:
nit: why did you have to cast this as any? What type is it expecting? Is blockTag supposed to be a string rather than a number (even if it is numerical)?

Reply:
typechain does not not have the correct teyp annotations for this function, so typescript fails when trying to pass the blockTag in, it doesnt recognize it as a parameter. This was the case at least when originally written

Reply:
going to try to remove it and see what happens, looks like its working now